File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int,int> freq;
        for (auto & n : arr) {
            if (freq.find(n) == freq.end()) {
                freq[n] = 0;
            }
            freq[n]++;
        }
        unordered_set<int> s;
        for (auto v : freq) {
            if (s.find(v.second) != s.end()) {
                return false;
            }
            s.insert(v.second);
        }
        return true;
    }
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        freq = defaultdict(int)
        for n in arr:
            freq[n] += 1
        s = set()
        for v in freq.values():
            if v in s:
                return False
            s.add(v)
        return True
